interface/forms/questionnaire_assessments/questionnaire_assessments.php

```diff
@@ -10,35 +10,71 @@
  * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
  */
 
+require_once(__DIR__ . "/../../../src/Common/Forms/CoreFormToPortalUtility.php");
+
+use OpenEMR\Common\Forms\CoreFormToPortalUtility;
+
+// block of code to securely support use by the patient portal
+$isPortal = CoreFormToPortalUtility::isPatientPortalSession($_GET);
+if ($isPortal) {
+    $ignoreAuth_onsite_portal = true;
+}
+$patientPortalOther = CoreFormToPortalUtility::isPatientPortalOther($_GET);
+
 require_once(__DIR__ . "/../../globals.php");
 require_once("$srcdir/user.inc");
 
 use OpenEMR\Common\Acl\AclMain;
 use OpenEMR\Common\Csrf\CsrfUtils;
 use OpenEMR\Core\Header;
+use OpenEMR\Services\QuestionnaireResponseService;
 use OpenEMR\Services\QuestionnaireService;
 
```

Comment:
ensure you have something that does not allow patients viewing other patient data. made a function for this in sdoh code at:
https://github.com/openemr/openemr/blob/master/interface/forms/sdoh/new.php#L34-L35